tests/osaka/eip7823_modexp_upper_bounds/test_eip_mainnet.py

```diff
@@ -0,0 +1,98 @@
+"""
+Mainnet marked execute checklist tests for
+[EIP-7823: ModExp Upper Bound](https://eips.ethereum.org/EIPS/eip-7823).
+"""
+
+from typing import Dict
+
+import pytest
+from execution_testing import (
+    Alloc,
+    StateTestFiller,
+    Transaction,
+)
+from execution_testing.base_types.base_types import Bytes
+from execution_testing.test_types.block_types import Environment
+
+from ...byzantium.eip198_modexp_precompile.helpers import (
+    ModExpInput,
+    ModExpOutput,
+)
+from ..eip7883_modexp_gas_increase.spec import Spec
+from .spec import ref_spec_7823
+
+REFERENCE_SPEC_GIT_PATH = ref_spec_7823.git_path
+REFERENCE_SPEC_VERSION = ref_spec_7823.version
+
+pytestmark = [pytest.mark.valid_at("Osaka"), pytest.mark.mainnet]
+
+
+@pytest.fixture
+def call_succeeds(modexp_expected: ModExpOutput) -> bool:
+    """Override `call_succeeds` to use the parametrized ModExpOutput value."""
+    return modexp_expected.call_success
+
+
+@pytest.mark.parametrize(
+    "modexp_input,modexp_expected",
+    [
+        pytest.param(
+            ModExpInput(
+                base=b"\x01" * Spec.MAX_LENGTH_BYTES,
+                exponent=b"\x00",
+                modulus=b"\x02",
+            ),
+            ModExpOutput(
+                call_success=True,
+                returned_data=Bytes(bytes.fromhex("01")),
```

Comment:
`ModExpOutput` is a pydantic model, it will handle hex strings just fine:
```suggestion
                returned_data="0x01",
```